spec/integrations/shared/base_render_examples.rb

```diff
@@ -50,6 +50,74 @@ def extract(field_name, object, _local_options, _options={})
     it('returns json derived from a custom extractor') { should eq(result) }
   end
 
+  context 'Given blueprint has ::field with a :if argument' do
+    variants = %i[proc method].product([true, false])
+
+    variants.each do |type, value|
+      context "#{type} returning #{value}" do
+        let(:if_value) { value }
+        let(:unless_value) { false }
+        let(:field_options) { {} }
+        let(:local_options) { { x: 1, y: 2 } }
+        let(:if_proc) { ->(_obj, _local_opts) { if_value } }
+        let(:unless_proc) { ->(_obj, _local_opts) { unless_value } }
+        let(:blueprint) do
+          f_options = field_options
+
+          bp = Class.new(Blueprinter::Base) do
+            field :id
+            field :first_name, f_options
+          end
+          bp.instance_eval <<-RUBY, __FILE__, __LINE__ + 1
+            def self.if_method(_object, _options)
+              #{if_value}
+            end
+
+            def self.unless_method(_object, _options)
+              #{unless_value}
+            end
+          RUBY
+          bp
+        end
+
+        let(:result_with_first_name) do
+          %({"first_name":"Meg","id":#{obj_id}})
+        end
+        let(:result_without_first_name) { %({"id":#{obj_id}}) }
+        subject { blueprint.render(obj, local_options) }
+
+        before do
+          field_options[:if] = type == :method ? :if_method : if_proc
+        end
+
+        context 'and no :unless argument' do
+          if value
+            it { should eq(result_with_first_name) }
+          else
+            it { should eq(result_without_first_name) }
+          end
+        end
+
+        variants.each do |other_type, other_value|
+          context "and :unless is #{other_type} returning #{other_value}" do
```

Comment:
The current location of this context block makes it look like `:unless` depends on first having `:if` passed to `field`.  Both `:if` and `:unless` should be at the same level of context right?

I'm thinking of something like:

```
    Inside Rails project
      Given blueprint has ::field with a conditional argument
        Given the conditional is :if
           ... other contexts / specs
        Given the conditional is :unless
           ... other contexts / spces
    Outside Rails project
      ... repeat from above ...
```

Reply:
it actually is: it's specs for both :if & :unless passed. Added specs for :unless only clarity and fixed examples descriptions, now it looks like:
```
    Inside Rails project
      Given blueprint has ::field with a conditional argument
        Given the conditional is :if proc returning true
          and no :unless conditional
            serializes the conditional field
          and :unless conditional is proc returning true
            does not serialize the conditional field
          …
        Given the conditional is :unless proc returning true and no :if conditional
          does not serialize the conditional field
        Given the conditional is :if proc returning false
          and no :unless conditional
            does not serialize the conditional field
          and :unless conditional is proc returning true
          …
```